vcd/provider.go

```diff
@@ -108,6 +111,45 @@ func Provider() terraform.ResourceProvider {
 	}
 }
 
+// This is a global MutexKV for all resources
+var vcdMutexKV = mutexkv.NewMutexKV()
```

Comment:
Is this the best place to put the mutex functions?
I didn't notice before that they are in Provider. Should we leave `provider.go` clean and add the mutex functions in `config.go`?
@Didainius , @lvirbalas , please chime in.

Reply:
AWS was holding in provider, so I also choose there. Though could agree on config too.

Reply:
Both places look fine.

Reply:
At the end I moved to config, cause function type changed and more functions were added.